src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -924,12 +924,21 @@ void Aura::SetStackAmount(uint8 stackAmount)
     SetNeedClientUpdateForTargets();
 }
 
+uint32 Aura::GetMaxStackAmount() const
```

Comment:
Could you rename this to `CalcMaxStackAmount` for consistency with CalcMaxCharges and to make it clear that it does not simply read a cached field?